cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/templates/etl/batch/ETLBatchTemplate.java

```diff
@@ -140,8 +140,8 @@ private void validateAdapter(ETLStage source, ETLStage sink, List<ETLStage> tran
     throws IllegalArgumentException {
     if (transformList.size() == 0) {
       // No transforms. Check only source and sink.
-      if (!(TypeToken.of(batchSink.getKeyType()).isAssignableFrom(batchSource.getKeyType()) &&
-        TypeToken.of(batchSink.getValueType()).isAssignableFrom(batchSource.getValueType()))) {
+      if (!(isAssignable(batchSource.getKeyType(), batchSink.getKeyType()) &&
```

Comment:
Then why no  just comment out all these and bring it in when you fix it. Its confusing to have it on develop